helpers/calendar/ratios_and_utilization.py

```diff
@@ -72,13 +75,21 @@ def get_analytics_ratios_per_room(self, query, start, end):
                                                         app_events_list)
 
             ratio_object = RoomAnalyticsRatios().map_results_to_ratio_class(
-                                checkins,
-                                cancellations,
-                                calendar_events_list,
-                                app_events_list,
-                                room['name'])
+                checkins=checkins,
+                cancellations=cancellations,
+                calendar_events_list=calendar_events_list,
+                app_events_list=app_events_list,
+                room_name=room['name'],
+                room_id=room['room_id'])
 
             response.append(ratio_object)
+
+        if room_id:
```

Comment:
You could check this before appending the `ratio_object` in the response array above and return the object without necessarily using the for loop.